namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+Often, before an MSON DOM can be processed referenced _Named Types_ have to be resolved. Resolving references to _Named Types_ is tedious and error prone. As such parser can resolve references and produce a complete MSON DOM, that is a DOM that does not include unresolved references to other data structures. This is referred to as _reference expansion_ or simply _expansion_.
+
+In other words, an expanded element  is one that does not contain any _Identifier_ (defined bellow) referencing any other elements than those defined in JSON or MSON namespaces.
+
+The expanded DOM MUST, however, keep the track of what data structure was expanded and what from where.
+
+## Base Element
+
+In MSON, every data structure is a sub-type of another data structure, and, therefore, it is directly or indirectly derived from one of the MSON _Base Types_. This is expressed as an inheritance of elements in MSON DOM. Where the predecessor of an element is referred to as element's _Base Element_.
+
+Note: Not every MSON _Base Type_ is presented in JSON namespace primitive types and vice versa, see the table bellow:
+
+### JSON Namespace vs. MSON built-in types
+
+| JSON primitive |   JSON Namespace   | MSON Base Type |   MSON Namespace   |
+|:--------------:|:------------------:|:--------------:|:------------------:|
+|     boolean    |  JSON:Boolean Type |     boolean    |  MSON:Boolean Type |
+|     string     |  JSON:String Type  |     string     |  MSON:String Type  |
+|     number     |  JSON:Number Type  |     number     |  MSON:Number Type  |
+|      array     |   JSON:Array Type  |      array     |   MSON:Array Type  |
+|        -       |          -         |      enum      |   MSON:Enum Type   |
+|     object     |  JSON:Object Type  |     object     |  MSON:Object Type  |
+|      null      |   JSON:Null Type   |        -       |          -         |
+|        -       | JSON:Property Type |        -       | MSON:Property Type |
+
+
+# General Elements Definition
+
+General elements defined inside MSON namespaces but possibly reusable in another domain.
+
+## Identifier (Enum)
+
+Identifier of an element in the MSON namespace. The identifier MUST be either directly a `string` value or an `Id Element` element.
+
+### Members
+
+- (string)
+- (Id Element)
+
+## Id Element (Distinct Element)
+
+Element representing an identifier in the MSON namespace.
+
+### Properties
+
+- `element`: id (string, required, fixed)
+- `content` (string, required)
+
+## Distinct Element (Element)
+
+Abstract element representing an element that can be referenced or is a reference to another element.
+
+### Properties
+
+- `element` (Identifier, required) - identifier of this element's base element
+- `attributes`
+    - `id` (Identifier) - identifier of this element
+    - `ref` (Identifier) - reference to expanded element
```

Comment:
Add me to the list of people who don't get this. What do you mean by `expanded element`? Going by the below examples, it looks like we can use `element` to denote a base element. Then, why do we need `ref`?